helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java

```diff
@@ -521,6 +543,11 @@ public String create(final String path, Object data, final CreateMode mode)
    */
   public String create(final String path, Object datat, final List<ACL> acl, final CreateMode mode)
       throws IllegalArgumentException, ZkException {
+    return create(path, datat, acl, mode, null);
+  }
+
+  private String create(final String path, Object datat, final List<ACL> acl, final CreateMode mode,
+      final String sessionId) throws IllegalArgumentException, ZkException {
```

Comment:
Right, I had the same idea. On the other side, I realized that we had metrics recording later in create() and retryUntilConnected(). If we check session id earlier, that means createEphemeral fails and we also need to record metrics. Thus adds more code to this early checking block. At the same time, I didn't think checking in retryUntilConnected() takes significant overhead (it will create a callable object, of course). So I decided not to do early check.